openedx/core/djangoapps/content_libraries/api/blocks.py

```diff
@@ -493,34 +501,91 @@ def _import_staged_block_as_container(
 
     title = _title_from_olx_node(olx_node)
 
-    # Start an atomic section so the whole paste succeeds or fails together:
-    with transaction.atomic():
-        container = create_container(
-            library_key=library_key,
-            container_type=ContainerType.Unit,
-            slug=None,  # auto-generate slug from title
-            title=title,
-            user_id=user.id,
-        )
-        new_child_keys: list[LibraryUsageLocatorV2] = []
-        for child_node in olx_node:
+    container = create_container(
+        library_key=library_key,
+        container_type=ContainerType.from_source_olx_tag(olx_node.tag),
+        slug=None,  # auto-generate slug from title
+        title=title,
+        user_id=user.id,
+    )
+
+    # Keep track of which blocks were copied from the library, so we don't duplicate them
+    if copied_from_map is None:
+        copied_from_map = {}
+
+    # Handle children
+    new_child_keys: list[LibraryUsageLocatorV2 | LibraryContainerLocator] = []
+    for child_node in olx_node:
+        childIsContainer = child_node.tag in ["vertical", "sequential", "chapter"]
```

Comment:
NIT: We could create a helper function like `isContainer` and use it here and other places like in line 616.